### ISSUE TITLE
Solution soundness bug on QF_S formula 

### ISSUE THREAD

**Issue by muchang** [CVC4/CVC4#5428]
Hi, for this formula, CVC4 gives an incorrect answer:

```
[634] % z3release small.smt2
unsat
[635] % cvc4 small.smt2
sat
[636] % 
[636] % cat small.smt2
(set-logic QF_S)
(assert (str.in_re "" (re.diff (re.* re.allchar) re.allchar (re.* re.allchar))))
(check-sat)
[637] %
```

OS: Ubuntu 18.04
Commit: 80930ce

**Comment by ajreynol** [comment]
This is incorrect syntax that somehow cvc4 (and z3?) is not catching.  After my PR, we respond:
```
(error "Parse Error: issue5428.smt2:2.78: Invalid kind 'REGEXP_DIFF', expected Terms with kind REGEXP_DIFF must have at least 2 children and at most 2 children (the one under construction has 3)")
```

**Comment by ajreynol** [comment]
Indeed, `re.diff` is left associative.  Internally, cvc4 was not treating it as left associative nor properly checking for cases where it was applied 2+ arguments internally.  This is fixed in #5430.
